Fix Area2D instancing causes unexpected effects on RigidBody2D

voidiii opened the pull request:
The variable `has_space_override `is incorrectly modified causing the destructor unable to remove areas from the list properly.
While `result `and `colliding `are both true, `has_space_override `should not change.
After the `pre_solve`, `process_collision `should be set to false. I think it makes more sense than changing it in the setup method.

*Bugsquad edit:*
- Fixes #81550.
- See also alternative solution in #81434.

Chaosus commented:
Thanks for contribution! Commits need to be squashed according to workflow (see https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html)

voidiii committed:
Fix Area2D instancing causes unexpected effects on RigidBody2D Issue#81550

The variable has_space_override is incorrectly modified and causing the destructor unable to remove area from the list properly.
While result and colliding are both true, has_space_override should not change.
After the pre_solve, process_collision should be set to false. I think it makes more sense than changing it in the setup method.

voidiii commented:
> Thanks for contribution! Commits need to be squashed according to workflow (see https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html)

OK I think I fixed it. How is it now?

voidiii commented:
Sry for the trouble, it is my first time

NolanDC commented:
I've got pull request up for this here: https://github.com/godotengine/godot/pull/81434 although it's still awaiting review.

voidiii commented:
> I've got pull request up for this here: https://github.com/godotengine/godot/pull/81434 although it's still awaiting review.

oh i didnt see yours. I will close this one then.

YuriSizov commented:
Closing per comment above (and as being superseded by https://github.com/godotengine/godot/pull/82961).
